ddtrace/internal/writer.py

```diff
@@ -197,6 +198,7 @@ def __init__(
             interval=processing_interval, exit_timeout=shutdown_timeout, name=self.__class__.__name__
         )
         self.agent_url = get_trace_url() if agent_url is None else agent_url
+        verify_url(self.agent_url)
```

Comment:
This is too late in the process IMHO.
You are checking a user-supplied value, check it when the user supplies it, not way down the road when "code" is interacting with you.

Reply:
Updated, it's not super pretty. Let me know what you think 🙂